### ISSUE TITLE
Foreground color of highlighted zsh autocompletion entry always black

### ISSUE THREAD

**Issue by synonym24** [gnunn1/tilix#1364]
My problem is shown in these screenshots:
![black-on-white](https://user-images.githubusercontent.com/3665667/38526887-6f972488-3c58-11e8-8e46-0bbe6415d227.png)
![red-on-white](https://user-images.githubusercontent.com/3665667/38526919-8e6e5372-3c58-11e8-93d6-bce3574910a0.png)
![yellow-on-green](https://user-images.githubusercontent.com/3665667/38526922-90d2ff28-3c58-11e8-8241-4ff908f27f8a.png)

It doesn't matter which color I choose as the foreground or background color, the text color of the highlighted entry of the zsh autocompleted entry is always black.

I checked it in the gnome terminal against, there it the text color of the highlighted entry is the background color from the settings.

Is this a bug in Tilix? Did I have something misconfigured?

**Comment by gnunn1** [comment]
If you set the highlight colors explicitly does it work? Just wondering if gnome-terminal is setting them when not set otherwise.

![screenshot from 2018-04-10 09-15-11](https://user-images.githubusercontent.com/9358238/38558987-ced95c9e-3c9f-11e8-923b-494898d49185.png)

**Comment by synonym24** [comment]
No, enabling and setting the Highlight colors doesn't have an affect to that. These setting only applies to selections with the mouse for me.

**Comment by egmontkob** [comment]
Probably zsh asks for black. Does changing the Black color change what you see?

I have no idea at the moment how Tilix and gnome-terminal could behave differently.

It would be nice to see what exactly zsh prints. Most likely the "reverse" attribute is involved. Could you please run `script`, repeat the steps that produce the bug, exit the shell launched by script, and attach the resulting `typescript` file? Please repeat the same steps with gnome-terminal too, maybe zsh prints something else there.

**Comment by synonym24** [comment]
Changing both of the black colors doesn't have an effect on that issue. But I can see that this settings has affect to other entries.

[scripts.zip](https://github.com/gnunn1/tilix/files/1900397/scripts.zip)

**Comment by egmontkob** [comment]
Thanks; the two typescript files are essentially the same. For inverse text, only the inverse attribute (SGR 7) is used, the colors are at their defaults. I bet the same issue is reproducible with this simple command:

    echo -e '\e[7mInverse text\e[27m'

Can you confirm this?

Do you have screen, tmux, byobu, anything similar running?

If not then this pretty much sounds like a VTE bug, maybe triggered by the way Tilix calls its API (maybe along the lines of the still unresolved https://bugzilla.gnome.org/show_bug.cgi?id=781369), maybe even related to transparency. I cannot reproduce the issue though.

What's your VTE version? (`echo $VTE_VERSION`)

**Comment by synonym24** [comment]
Yes, the echo reproduces this behavior.
No, I don't use screen or anything similar and I'm using VTE version 5200.

Here is a screenshot of the requested infos:
![image](https://user-images.githubusercontent.com/3665667/38648546-13cbb3d0-3df2-11e8-8612-8802a2e40fa3.png)

Thank you for the clarification. Hopefully this bug, at least for me it's a bug, will be addressed soon.

**Comment by egmontkob** [comment]
Well, this latter screenshot isn't the same as the first one. There it was unreadable (black on black), now it's black on dark gray. I'm getting even more confused. Did you change your setting since then, or does this little example really produce something different than your zsh?

In the first three screenshots, the background of the highlighted entry isn't the same as the chosen foreground color, it's always 77% of the intensity thereof, which is probably related to the transparency slider seemingly standing at about 23%. However, I'm wondering why your terminal isn't transparent. Do you have a totally black background? Or the window (or desktop background) underneath being black? Or transparency not working due to some reason? It might be relevant, I don't know.

(Why would you close this issue when we still don't have a clue what's going on? Bugs hardly ever get magically fixed on their own, it's the developers who reproduce and fix them, and in order to do that they need to keep track of the pending issues.)

**Comment by synonym24** [comment]
Let me clear up your confusion.

First, yeah I changed the forecolor. The initial three screenshots were made especially to visualize this issue. I also reverted to standard settings (Gnome Shell theme, application theme, even the mouse cursor settings and icons) to be sure it isn't related to some settings of mine.
Yes I took that on a black background. I didn't want to irritate you guys with a background that could interfere with the selected colors.
Changing the foreground color was necessary for me because I have noticed, that I use the auto completion all the time and it's not fun to look at a black bar and always searching the text of the currently selected item.

I too opened the three screenshots in Gimp and compared the colors by selecting each with the picker tool. For me the selected foreground color from the settings window and the background of the selected entry (highlighting of the entry Templates) are the same. First image color hex code both #000000 (black), second both #ef2929 (red), third both #fce94f (yellow).

Here a screenshot were I checked the yellow foreground color (the third screenshot in the inital post) in Gimp:
![image](https://user-images.githubusercontent.com/3665667/38707425-998f44ea-3eb1-11e8-856f-20170074805a.png)
Normally I'm using the Gnome Shell theme and application theme Arrongin which I both selected via the gnome tweak tool which you can see here. I've only been using this themes for a short time, that's why I didn't even notice the bug in the first place.

The screenshot in my reply from Apr 12, 2018, 1:47 AM GMT+2 was taken with gnome-screenshot and the following settings:
![image](https://user-images.githubusercontent.com/3665667/38706981-99a276ca-3eaf-11e8-8f10-ef2770d00000.png)
With "grab the current window" the background or any window below the current window is ignored.

To be sure, here is another screenshot with the same requested info's. Tilix is here above Firefox. As you can see with that shot, transparency is working great for me.
![image](https://user-images.githubusercontent.com/3665667/38707783-18753afc-3eb3-11e8-9ff5-246f5f4314f7.png)

Regarding my question for closing this issue:
From your answer from Apr 11, 2018, 10:54 PM GMT+2 I assume the problem was related to vte, so I asked if I could/should close this.

**Comment by gnunn1** [comment]
I can reproduce the problem in both bash and zsh. When you echo the inverse text, the text color is always black for me regardless of what is set in tilix for foreground/background colors so only the foreground color is getting inversed.

I'm not sure what is going on but am having a look at it. gnome-terminal is working fine on my machine so it does seem specific to tilix.

**Comment by gnunn1** [comment]
So I found the problem, the issue is that in the old way I handled badges I had to set the VTE background color to be clear and artificially paint it. The problem is that inverse text would then use this clear background color as the text color. With the new VTE clear background method this is no longer required and could be removed.

I suspect this was never reported previously because relatively few people were using my VTE patches for badge and trigger support.

Here's a screenshot with the fix showing badges working as well.

![screenshot from 2018-04-12 20-20-37](https://user-images.githubusercontent.com/9358238/38710676-9f6d08ce-3e8f-11e8-92f6-8aea6067abca.png)

**Comment by egmontkob** [comment]
Thanks guys! :)
